### PR TITLE
Edit task status

### PR DESCRIPTION
**Pull request by ninest**
This PR depends on #789 being merged.

## Changes

_Explanation of changes goes here_

## Notes

_Any other notes go here_

## Test Cases

- Case A
- Edge case
- ...

## Screenshots

_If you made UI changes you must post a screenshot of the whole page for each change in 1) a normal sized window and 2) the smallest possible window_

_If you did any manual testing (e.g., with Postman), put screenshots of the http request and before and after of the db_

_If none of this applies, you can delete this section_

## To Do

_Any remaining things that need to get done_

- [ ] item 1
- [ ] ...

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [ ] All commits are tagged with the ticket number
- [ ] No linting errors / newline at end of file warnings
- [ ] All code follows repository-configured prettier formatting
- [ ] No merge conflicts
- [ ] All checks passing
- [ ] Screenshots of UI changes (see Screenshots section)
- [ ] Remove any non-applicable sections of this template
- [ ] Assign the PR to yourself
- [ ] No `yarn.lock` changes (unless dependencies have changed)
- [ ] Request reviewers & ping on Slack
- [ ] PR is linked to the ticket (fill in the closes line below)

Closes # (issue #)

### CODE REVIEW

**Review by anthonybernardi — COMMENTED** [review]
awesome work! only thing is i want to make sure the permissions on who can edit a task's status is correct. other than that everything looks great!

**Comment by ninest** [comment]
Can edit a task if

- Any admin, app admin
- task's createdBy
- wbsElement's or projectLead, projectManager
- 	if project exists and project is on a team, team's leader
- 	if work package exists, go into project, and make same check

File: src/backend/src/utils/tasks.utils.ts [hunk 1]
```diff
@@ -14,3 +15,93 @@ export const convertTaskStatus = (status: Task_Status): TaskStatus =>
...
+  if (user.role === Role.ADMIN || user.role === Role.APP_ADMIN) {
+    return true;
+  }
+
+  const task = await prisma.task.findUnique({
+    where: { taskId },
+    select: {
+      createdByUserId: true,
+      wbsElement: {
+        select: {
```
Comment (ninest) [inline src/backend/src/utils/tasks.utils.ts:29]: I created this util function to check if a user has permission to edit a task. Should I create a separate test for this function? Currently I onnly have tests for `TasksService.editTaskStatus` which calls `hasPermissionToEditTask`

File: src/backend/src/utils/tasks.utils.ts [hunk 2]
```diff
@@ -14,3 +15,93 @@ export const convertTaskStatus = (status: Task_Status): TaskStatus =>
...
+  }
+  if (task.wbsElement.projectManagerId === user.userId) {
+    return true;
+  }
+
+  // Check if the user is the project leader or on the project team
+  if (task.wbsElement.project?.team?.leaderId === user.userId) {
+    return true;
+  }
+  const projectTeamUserIds = task?.wbsElement.project?.team?.members?.map((user) => user.userId);
```
Comment (anthonybernardi) [inline src/backend/src/utils/tasks.utils.ts:90]: i dont think we should allow all members of the team to edit it, just the team lead

File: src/backend/src/utils/tasks.utils.ts [hunk 3]
```diff
@@ -14,3 +15,93 @@ export const convertTaskStatus = (status: Task_Status): TaskStatus =>
...
+  if (projectTeamUserIds?.includes(user.userId)) {
+    return true;
+  }
+
+  // Do the same thing, but for the work package's project
+  if (task.wbsElement.workPackage?.project?.team?.leaderId === user.userId) {
+    return true;
+  }
+  const workPackageProjectTeamMemberUserIds = task?.wbsElement.workPackage?.project?.team?.members?.map(
```
Comment (anthonybernardi) [inline src/backend/src/utils/tasks.utils.ts:99]: same thing

File: src/backend/src/utils/tasks.utils.ts [hunk 4]
```diff
@@ -14,3 +15,93 @@ export const convertTaskStatus = (status: Task_Status): TaskStatus =>
     IN_PROGRESS: TaskStatus.IN_PROGRESS,
     DONE: TaskStatus.DONE
   }[status]);
+
+export const hasPermissionToEditTask = async (user: User, taskId: string): Promise<boolean> => {
+  if (user.role === Role.ADMIN || user.role === Role.APP_ADMIN) {
+    return true;
+  }
+
+  const task = await prisma.task.findUnique({
```
Comment (anthonybernardi) [inline src/backend/src/utils/tasks.utils.ts:24]: only other thing to check is if the user is one of the assignees of the task

File: src/backend/src/utils/tasks.utils.ts (hunk 1 above)
Comment (anthonybernardi) [reply src/backend/src/utils/tasks.utils.ts:29]: probably a good idea if you don't mind doing it really quick but if it takes too long we can skip it tbh

**Review by anthonybernardi — CHANGES_REQUESTED** [review]
nice!
